homeassistant/components/backup/manager.py

```diff
@@ -235,24 +243,37 @@ class BackupManager(BaseBackupManager[Backup]):
     def __init__(self, hass: HomeAssistant) -> None:
         """Initialize the backup manager."""
         super().__init__(hass=hass)
-        self.backup_dir = Path(hass.config.path("backups"))
-        self.loaded_backups = False
+        self.temp_backup_dir = Path(hass.config.path("tmp_backups"))
 
     async def async_upload_backup(self, *, slug: str, **kwargs: Any) -> None:
-        """Upload a backup."""
-        await self.load_platforms()
-
+        """Upload a backup to all agents."""
         if not self.backup_agents:
             return
 
         if not (backup := await self.async_get_backup(slug=slug)):
             return
 
+        local_agent = self.local_backup_agents[LOCAL_AGENT_ID]
+        await self._async_upload_backup(
+            backup=backup,
+            agent_ids=list(self.backup_agents.keys()),
+            # TODO: This should be the path to the backup file
+            path=local_agent.get_backup_path(slug),
```

Comment:
This should be done for all local agents.

Reply:
The method is removed, so this is no longer relevant